File: python/reverse_string.py
#문자열 뒤집기

'''
answer 1

def solution(my_string):
    answer = ''
    
    answer = my_string[::-1]
    
    return answer
'''

'''
answer 2

def solution(my_string):
    answer = ''
    
    for i in range(len(my_string) - 1, -1, -1) :
        answer += my_string[i]
    
    return answer

'''

'''
answer 3

def solution(my_string):
    answer = ''
    my_list = list(my_string)
    my_list.reverse()
    
    answer = ''.join(my_list)
    
    return answer
  '''

'''
answer 4

def solution(my_string):

    answer = ''.join(reversed(my_string))
    
    return answer

'''

File: cpp/lv0/reverse_string.cpp
// 문자열 뒤집기
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

/* Answer 1.

string solution(string my_string) {
    string answer = "";
    
    reverse(my_string.begin(), my_string.end());
    answer = my_string;
    
    return answer;
} */


/* Answer 2.

string solution(string my_string) {
    reverse(my_string.begin(), my_string.end());
    
    return my_string;
} */


/* Answer 3.

string solution(string my_string) {
    string answer = "";
    
    for(int i = my_string.size() - 1; i >= 0; i--)
    {
        answer += my_string[i];
    }
    
    return answer;
} */
